examples/html-fragment/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"embed"
+	"flag"
+	"fmt"
+	"html/template"
+	"io/fs"
+	"log"
+	"net"
+	"net/http"
+	"os"
+
+	"github.com/olivere/vite"
+)
+
+//go:embed all:static
+var static embed.FS
+
+//go:embed index.tmpl
+var goIndex embed.FS
+
+func main() {
+	var (
+		isDev = flag.Bool("dev", false, "run in development mode")
+	)
+	flag.Parse()
+
+	mux := http.NewServeMux()
+
+	// Serve assets that Vite would treat as 'public' assets.
+	//
+	// In this example, the 'static' directory is used as a replacement for
+	// Vite's default 'public' folder, and 'publicDir' is disabled in the Vite
+	// config. We're using the 'static' directory to achieve similar
+	// functionality, but available to the Go backend and Vite.
+	//
+	// To use a static asset in our Vite frontend, we import it like this:
+	//
+	// import viteLogo from '/static/vite.svg'
+
+	var staticFileServer http.Handler
+	if *isDev {
+		staticFileServer = http.FileServer(http.Dir("static"))
+	} else {
+		staticFS, err := fs.Sub(static, "static")
+		if err != nil {
+			panic(err)
+		}
+		staticFileServer = http.FileServer(http.FS(staticFS))
+	}
+
+	mux.Handle("/static/", http.StripPrefix("/static/", staticFileServer))
+
+	// Serve Vite-managed assets from the Go backend, accommodating both
+	// development and production environments.
+	//
+	// Usage in Vite remains the same as in a standard Vite setup. The Go backend
+	// will serve the assets from the correct location based on the environment.
+
+	var viteAssetsFileServer http.Handler
+	var viteAssetsSrcAttribute string // vite changes this path, depending on dev or prod
+	if *isDev {
+		viteAssetsSrcAttribute = "/src/assets/"
+		viteAssetsFileServer = http.FileServer(http.Dir("src/assets"))
+	} else {
+		viteAssetsFS, err := fs.Sub(static, "static/dist/assets")
+		if err != nil {
+			panic(err)
+		}
+		viteAssetsFileServer = http.FileServer(http.FS(viteAssetsFS))
+		viteAssetsSrcAttribute = "/assets/"
+	}
+
+	mux.Handle(viteAssetsSrcAttribute, http.StripPrefix(viteAssetsSrcAttribute, viteAssetsFileServer))
+
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// viteFS represents the filesystem used for serving Vite assets.
+		//
+		// This variable is used to configure the Vite integration, ensuring that
+		// the correct assets are served based on the current environment.
+		var viteFS fs.FS
+		if *isDev {
+			viteFS = os.DirFS(".")
```

Comment:
Same again, I'm not sure if we need to serve any Vite files in dev mode?

Reply:
I'll be sure to test.

Reply:
https://github.com/olivere/vite/pull/10#discussion_r1739042855